src/main/java/com/synopsys/integration/jenkins/scan/service/scm/BitbucketRepositoryService.java

```diff
@@ -36,9 +36,9 @@ public Bitbucket fetchBitbucketRepositoryDetails(Jenkins jenkins, Map<String, Ob
                 BitbucketSCMSource bitbucketSCMSource = (BitbucketSCMSource) scmSource;
 
                 String bitBucketToken = (String) scanParameters.get("bitbucket_token");
-                if(bitBucketToken == null) {
-                    bitBucketToken = SCMRepositoryService.getCredentialsToken(bitbucketSCMSource.getCredentialsId());
-                }
+//                if(bitBucketToken == null) {
```

Comment:
We should add a private method like `getBitbucketTokenFromGlobalConfig()`  here at `BitbucketRepositoryService` 

```
public String getBitbucketTokenFromGlobalConfig() {
        ScannerGlobalConfig config = GlobalConfiguration.all().get(ScannerGlobalConfig.class);
        if (config != null && !Utility.isStringNullOrBlank(config.getBitbucketToken())) {
            return config.getBitbucketToken().trim();
        }
        return null;
    }
```

then 

```
if (bitbucketToken == null) {
   bitbucketToken = getBitbucketTokenFromGlobalConfig()
}
```

**_N.B. Here there may be test failure in `BitbucketRepositoryServiceTest`_**

Reply:
Thanks for the code snippet.